txtorcon/torcontrolprotocol.py

```diff
@@ -641,6 +641,19 @@ def _safecookie_authchallenge(self, reply):
         client_hash_hex = base64.b16encode(client_hash)
         return self.queue_command('AUTHENTICATE %s' % client_hash_hex)
 
+    
+    def _read_cookie(self, cookiefile):
+        """
+        Open and read a cookie file
+        :param cookie: Path to the cookie file
+        """
+        self.cookie_data = open(cookiefile, 'r').read()
+        if len(self.cookie_data) != 32:
```

Comment:
This should be 'rb' since we're expecting bytes. And although this isn't consistent at all in txtorcon yet, I'd like to keep data like this as "private" attribute, so ``self._cookied_data`` please :)